gobblin-utility/src/main/java/gobblin/util/HadoopUtils.java

```diff
@@ -110,6 +113,7 @@ public static Configuration newConfiguration() {
     // Add a new custom filesystem mapping
     conf.set("fs.sftp.impl", "gobblin.source.extractor.extract.sftp.SftpLightWeightFileSystem");
     conf.set("fs.sftp.impl.disable.cache", "true");
+    conf.set("fs.adl.impl", "org.apache.hadoop.fs.adl.AdlFileSystem");
```

Comment:
Yes, it supposed to pick it up in runtime without this configuration, but it's not working with AdlFileSystem. I looked at their jar and they seemed misplaced service information under "META-INF/org.apache.hadoop.fs.FileSystem", not in "META-INF/services/org.apache.hadoop.fs.FileSystem"